src/Nethermind/Nethermind.Crypto/ProtectedData.T.cs

```diff
@@ -43,13 +43,13 @@ public ProtectedData(byte[] data, ICryptoRandom? random = null, ITimestamper? ti
         private void Protect(byte[] data)
         {
             _entropy = _random.GenerateRandomBytes(_random.NextInt(EntropyMaxLength - EntropyMinLength) + EntropyMinLength);
-            _encryptedData = Protect(data, _entropy, DataProtectionScope.CurrentUser);
+            _encryptedData = Protect(data, _entropy);
```

Comment:
`DataProtectionScope.CurrentUser` is only available on Windows.  The code was setup in such a way that this was only *used* on Windows, but the compiler couldn't verify that.  I restructured things a little bit so that the compiler can now recognize that `DataProtectionScope.CurrentUser` is only available on Windows.

```
                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                {
                    return GetUserProtector(optionalEntropy);
                }
                else
                {
                    return GetMachineProtector(optionalEntropy);
                }
```
I also think that this makes it a bit more clear that you only get per-user protection on Windows, which has always been the case but now it is more obvious to the reader that that is the case.